### PR TITLE
backend-plugin-api: remove identity and token manager services

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This completely removes support for the identity and token manager services from the new backend system. More info on the impact and potential workarounds in the changesets.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -14,11 +14,52 @@
...
+  createServiceRef,
+} from '@backstage/backend-plugin-api';
+import {
+  DefaultIdentityClient,
+  IdentityApi,
+} from '@backstage/plugin-auth-node';
 
 const backend = createBackend();
 
+backend.add(
```
Comment (freben) [inline packages/backend/src/index.ts:32]: Are these here mostly for demonstration, but no longer used anywhere?

File: plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export class CatalogIdentityClient {
 
   constructor(options: {
     catalogApi: CatalogApi;
-    tokenManager: TokenManager;
```
Comment (freben) [inline plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts:48]: Just keeping things nonbreaking I presume

**Review by freben — APPROVED** [review]

File: packages/backend/src/index.ts (hunk 1 above)
Comment (Rugvip) [reply packages/backend/src/index.ts]: They're here mostly by mistake 😁, was just using it to verify what I was putting in the changeset

File: plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts (hunk 2 above)
Comment (Rugvip) [reply plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts:48]: Yep, in the new system we'll use the actual service, in the old we still end up at https://github.com/backstage/backstage/blob/b9dca423cdf8c50ecade8ec7392220fb263c7ea5/packages/backend-common/src/compat/auth/createLegacyAuthAdapters.ts#L105

**Pull request merged**